uui/components/tables/DataRowsContainer/DataRows.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useRef } from 'react';
+import { DataRowProps } from '@epam/uui-core';
+import css from './DataRowsContainer.module.scss';
+import { getChildrenAndRest } from './utils';
+
+interface GroupProps<TItem, TId> {
+    row: DataRowProps<TItem, TId>;
+    childrenRows: DataRowProps<TItem, TId>[];
+    renderRow: (props: DataRowProps<TItem, TId>) => React.ReactNode;
+    top: number;
+}
+
+export interface DataRowsProps<TItem, TId> {
+    rows: DataRowProps<TItem, TId>[],
+    renderRow: (props: DataRowProps<TItem, TId>) => React.ReactNode;
+    top: number;
+}
+
+function DataRowGroup<TItem, TId>({
+    row,
+    childrenRows,
+    renderRow,
+    top = 1,
+}: GroupProps<TItem, TId>) {
+    const rowRef = useRef<HTMLDivElement>();
+    const childrenPinnedTop = row.isPinned ? (top + (rowRef.current?.clientHeight ?? 0)) : top;
+    return ( 
+        <div className={ css.group } key={ row.rowKey }>
+            <div className={ row.isPinned ? css.stickyHeader : css.header } style={ { zIndex: row.depth + 10, top } } ref={ rowRef }>
+                {renderRow(row)}
+            </div>
+            {childrenRows.length > 0 && (
+                <div>
+                    <DataRows rows={ childrenRows } renderRow={ renderRow } top={ childrenPinnedTop } />
+                </div>
+            )}
+        </div>
+    );
+}
+export function DataRows<TItem, TId>({
```

Comment:
We need another name for this component. With just 'DataRows', I'm getting confused with DataRowsProps interface - it clashes with DataRowProps (with singular 'Row') - which have a different meaning.

Reply:
Done.